src/internal/RemoteConfig.d

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+
+    Helper methods to access the octod/github API
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module internal.RemoteConfig;
+
+// Fixing import detection
+static import internal.github.oauth;
+
+import octod.core;
+import octod.api.Repos;
+
+/// Helper methods to access github and local github configuration
+struct RemoteConfig
+{
+    /// Name of the application accessing github
+    string app_name;
+
+    /// Octo config
+    Configuration config;
+
+    /***************************************************************************
+
+        Create instance of RemoteConfig, initializes an octod config as well
+        and will ask the user to set one up if the config data can't be found
+
+        Params:
+            app_name = name of the app
+            base_url = base url to the remote server, either github or gitlab
+            assume_yes = bool, assume true for any questions
+
+    ***************************************************************************/
+
+    public this ( string app_name, string base_url, bool assume_yes )
+    {
+        import internal.git.helper;
+        import std.exception;
+
+        this.app_name = app_name;
+
+        immutable GitHubConfName = ".oauthtoken";
```

Comment:
I think we should make this forward-compatible and also check for `.github-oauthtoken` with an assumption that git-hub will also eventually becomes generic.